custom-functions/remove-null-fields/index.js

```diff
@@ -1,59 +1,124 @@
-const cLogger = C.util.getLogger("func:RemoveNullFields");
+const cLogger = C.util.getLogger("func:removeNullFields");
 
 exports.disabled = 0;
 exports.name = "Remove null fields";
-exports.version = "0.1.0";
+exports.version = "0.1.1";
 exports.group = "Custom Functions";
 
+/**
+ * Instance of the RemoveFields class used to determine if a field should be removed
+ * based on the provided configuration. Initialized in the init function.
+ * @type {RemoveFields}
+ */
 let shouldRemove;
+
+/**
+ * Indicates whether the function should be disabled based on the configuration.
+ * The function is disabled if both `removeNull` and `removeEmpty` are false.
+ * @type {boolean}
+ */
 let isDisabled;
 
+/**
+ * Initializes the RemoveNullFields function with the provided options.
+ * @param {Object} options - The options to initialize the function.
+ * @param {Object} options.conf - The configuration object.
+ * @param {boolean} options.conf.removeNull - Toggle to remove null fields.
+ * @param {boolean} options.conf.removeEmpty - Toggle to remove empty fields.
+ * @param {number} [options.pipeIdx] - The index of the pipeline.
+ * @param {string} [options.pid] - The pipeline ID.
+ * @param {string} [options.cid] - The worker process ID.
+ */
 exports.init = (options) => {
-  const conf = options?.conf ?? {};
-  const funcIdx = options?.pipeIdx ?? undefined;
-  const pipeId = options?.pid ?? undefined;
-  const wp = options?.cid ?? undefined;
-  const { removeNull = false, removeEmpty = false } = conf;
+  const {
+    conf = {},
+    pipeIdx: funcIdx = undefined,
+    pid: pipeId = undefined,
+    cid: wp = undefined,
+  } = options;
+  
+  const { removeNull = false, removeEmpty = true } = conf;
+
+  // Determine if the function should be disabled
   isDisabled = !removeNull && !removeEmpty;
 
-  cLogger.debug("initializing RemoveNullFields function", {
-    pipeId,
-    funcIdx,
-    wp,
-    conf,
-  });
-  shouldRemove = new RemoveFields({ removeNull, removeEmpty });
+  // Log the initialization details
+  cLogger.info("initializing RemoveNullFields function", { pipeId, funcIdx, wp, conf });
+
+  // Create an instance of RemoveFields based on the configuration
+  shouldRemove = new RemoveFields(removeNull, removeEmpty)
```

Comment:
Instead of passing a JSON Object here, and de-structuring in the class, just pass two fields.